echo_api.rb

```diff
@@ -29,19 +32,46 @@ def get_headers
 
 def echo_response
   request.body.rewind
-
-  content_type 'application/json'
-  JSON.pretty_generate(
-    method: request.request_method,
-    path: request.path,
-    args: request.query_string,
-    body: request.body.read,
-    headers: get_headers()
-  )
+  if request.accept?('application/json')
```

Comment:
In my testing, if you don't set the accept header sinatra will return `true`  for t his.